x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/other_type_name_parameter.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { UseField, TextField } from '../../../shared_imports';
+import { fieldValidators } from '../../../shared_imports';
+
+const { emptyField } = fieldValidators;
+
+/**
```

Comment:
I love the idea, much better!

Why don't we do the same with the `otherTypeJson` which is not either a mapping parameter and shouldn't be in the PARAMETERS_DEFINITION.

I would though declare the field config object outside the component.

```js
const config: FieldConfig = { ... };

export const OtherTypeJsonParameter = () => (
  <UseField
    path="otherTypeJson"
    config={config}
    component={JsonEditorField}
  />
);
```